### PR TITLE
qns

### PR DESCRIPTION
**Pull request by yehjxraymond**

### CODE REVIEW

File: src/components/common/DomListener.stories.mdx [hunk 1]
```diff
@@ -35,6 +35,8 @@ When used as a root component, it will fire an action for any dom mutation and r
 </Preview>
 
 However be mindful that using it as a nested component will trigger DOM mutations only for children. In the following example clicking on buttons won't fire any event as the component is observing only the `+` button and no changes are made there. Resizing the window still works.
+^ This is a weird explanation. Were you trying to say that triggering an event without DOM changes will not fire `onUpdate`?
```
Comment (Nebulis) [inline src/components/common/DomListener.stories.mdx]: not exactly, this component only listen for children modification and not for document modification:

```jsx
<SomeComponent />
<DomListener><AnotherComponent></DomListener>
```

any DOM changes happening in `SomeComponent` wont be caught by `DomListener`.

File: src/components/common/DomListener.tsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import ReactDOM from "react-dom";
 // no tests because js-dom doesn't support MutationObserver
 // https://github.com/jsdom/jsdom/issues/639
 
+// Why do we differentiate `resize` from `domChange`? `domChange` can be anything liek att, childList, subtree & characterData also
```
Comment (Nebulis) [inline src/components/common/DomListener.tsx]: no specific reason as we listen to 2 different kind of event I thought would be nice to give back what's causing the function to be called

File: src/components/common/DomListener.tsx [hunk 3]
```diff
@@ -35,9 +36,11 @@ export class DomListener extends Component<DomListenerProps> {
     if (rootNode) {
       this.observer.observe(rootNode, { attributes: true, childList: true, subtree: true, characterData: true });
       // need an initial trigger, make it run on next tick :shrug:
+      // Why does it need to happen on the next tick?
       setTimeout(this.onUpdateHeight);
     }
-    window.addEventListener("resize", this.onResize); // ResizeObserver not good enough
+    // MutationObserver does not listen to resize events
```
Comment (Nebulis) [inline src/components/common/DomListener.tsx]: you want to add this comment ?

File: src/components/common/DomListener.stories.mdx (hunk 1 above)
Comment (yehjxraymond) [reply src/components/common/DomListener.stories.mdx]: That's intuitive right? Idk... 
Still this storybook component looks like a negative test to me. If it's a negative test I thought it should be in test and not showcased in storybook. Was very confused when I saw the component (esp when clicking wont work) until I read the code.

File: src/components/common/DomListener.tsx (hunk 2 above)
Comment (yehjxraymond) [reply src/components/common/DomListener.tsx]: Hmm im not sure if there's any value in this. Considering that you have to go all the way to curry the function without adding `curry` at the front/back. That's plain blasphemy.

File: src/components/common/DomListener.tsx [hunk 4]
```diff
@@ -35,9 +36,11 @@ export class DomListener extends Component<DomListenerProps> {
     if (rootNode) {
       this.observer.observe(rootNode, { attributes: true, childList: true, subtree: true, characterData: true });
       // need an initial trigger, make it run on next tick :shrug:
+      // Why does it need to happen on the next tick?
```
[the remark this answers is not in the document]
Comment (yehjxraymond) [reply src/components/common/DomListener.tsx]: Hmm I tried running it directly and it didnt not work. Also considering that this is already in `componentDidMount`, it's already in separate tick from the render function right?

File: src/components/common/DomListener.tsx (hunk 3 above)
Comment (yehjxraymond) [reply src/components/common/DomListener.tsx]: Yea.. `ResizeObserver` isnt a thing..

File: src/components/common/DomListener.stories.mdx (hunk 1 above)
Comment (Nebulis) [reply src/components/common/DomListener.stories.mdx]: as we discussed long time ago => will refactor by separating MutationObserver and window.resize listener

File: src/components/common/DomListener.tsx (hunk 2 above)
Comment (Nebulis) [reply src/components/common/DomListener.tsx]: see #2

File: src/components/common/DomListener.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Nebulis) [reply src/components/common/DomListener.tsx]: works as expected, will keep this until a better solution has been found out

File: src/components/common/DomListener.tsx (hunk 3 above)
Comment (Nebulis) [reply src/components/common/DomListener.tsx]: It is :p not well supported though

**Pull request closed without merging**